src/server/frontend_xwayland/xcb_connection.h

```diff
@@ -90,6 +91,46 @@ class XCBConnection
         std::atomic<xcb_atom_t> mutable atom{XCB_ATOM_NONE};
     };
 
+    struct ErrorPtr
+    {
+        ~ErrorPtr()
+        {
+            if (error)
+            {
+                free(error);
+            }
+        }
+
+        operator xcb_generic_error_t**() { return &error; }
+
+        xcb_generic_error_t* error{nullptr};
+    };
```

Comment:
Looks as though ErrorPtr has incorrect CopyAssign semantics. (Is this a case for `UniqueCPtr`?)

Reply:
I don't think you can get mutable access to the underlying pointer of a `unique_ptr`, right? The error has to be initialized as null, and then optionally set by the XCB function.